### ISSUE TITLE
Combination of recursive class, ABC, dataclass, and Generic gives error

### ISSUE THREAD

**Issue by ianliu** [microsoft/pyright#2141]
**pyright version**: 1.1.159
**environment**: coc-pyright with neovim v0.5.0

Consider the following code snippet:

```python
from __future__ import annotations
from dataclasses import dataclass
from abc import ABC
from collections.abc import Sequence
from typing import Generic, TypeVar

A = TypeVar("A")

@dataclass
class Foo(ABC, Generic[A]):
    children: Sequence[Foo[A]]
    #                  ^^^^^^ Class definition for "Foo" depends on itself
```

If I remove any of `ABC`, `Generic`, or `dataclass`, it type checks, but as presented, I get a _Class definition for "Foo" depends on itself_.

I think it should type check, since `Sequence` is covariant, and it should accept subclasses of `Foo` there.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.160, which I just published. It will also be included in the next release of pylance.
